config/base.js

```diff
@@ -82,7 +82,8 @@ const SCHEMA_CONFIG = Joi.object()
         childPipelines: SCHEMA_CHILD_PIPELINES.when('template', { is: Joi.exist(), then: Joi.forbidden() }),
         stages: SCHEMA_STAGES.when('template', { is: Joi.exist(), then: Joi.forbidden() }),
         subscribe: SCHEMA_SUBSCRIBE.when('template', { is: Joi.exist(), then: Joi.forbidden() }),
-        parameters: Parameters.parameters.default({}).when('template', { is: Joi.exist(), then: Joi.forbidden() })
+        parameters: Parameters.parameters.default({}).when('template', { is: Joi.exist(), then: Joi.forbidden() }),
+        templateVersionId: Joi.number().integer().positive().optional().allow(null)
```

Comment:
Is `templateVersionId` a valid field to have on the config?
This this shouldn't be allowed to be specified in `screwdriver.yaml`.

Reply:
it's needed because it will be added in config parser after we merge the template

Reply:
since it's a field that purely just provides information to user, i think there's any harm. do you have any suggestion?

Reply:
This could potentially be an input from the user even though it is not intended/recommended by design. If we add this to config schema, it would mean that `templateVersionId` is a valid field to be specified in the pipeline configuration `screwdriver.yaml`.

Suggestions:
1. Have 2 variations of the schema: one with `template` and other with `templateId`. Use the first one validation before merging pipeline template and second one for validation post merge.
2. Make `template` and `templateId` at least mutually exclusive. Leave a note for `templateId` that it is not expected to be configured in `screwdriver.yaml`.

I would incline towards option 1 as it is more cleaner and completely blocks the use from specifying it in `screwdriver.yaml`.

Considering the same problem exists for job template, option 2 could be considered for short term.